src/devices/Ft4222/Ft4222I2c.cs

```diff
@@ -0,0 +1,145 @@
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, ref _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the clock
+            FtClockRate ft4222Clock = FtClockRate.Clock24MHz;
+
+            ftStatus = FtFunction.FT4222_SetClock(_ftHandle, ft4222Clock);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed set clock rate {ft4222Clock} on device: {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the device as I2C Master
+            ftStatus = FtFunction.FT4222_I2CMaster_Init(_ftHandle, 400);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to initialize I2C Master mode on device: {DeviceInformation.Description}, status: {ftStatus}");
+        }
+
+        /// <summary>
+        /// The connection settings of a device on an I2C bus. The connection settings are immutable after the device is created
+        /// so the object returned will be a clone of the settings object.
+        /// </summary>
+        public override I2cConnectionSettings ConnectionSettings => _settings;
+
+        /// <summary>
+        /// Reads data from the I2C device.
+        /// </summary>
+        /// <param name="buffer">
+        /// The buffer to read the data from the I2C device.
+        /// The length of the buffer determines how much data to read from the I2C device.
+        /// </param>
+        public override void Read(Span<byte> buffer)
+        {
+            byte[] buff = new byte[buffer.Length];
+            ushort byteRead = 0;
+            var ftStatus = FtFunction.FT4222_I2CMaster_Read(_ftHandle, (ushort)_settings.DeviceAddress, buff, (ushort)buff.Length, ref byteRead);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"{nameof(Read)} failed to read, error: {ftStatus}");
+
+            buff.CopyTo(buffer);
+        }
+
+        /// <summary>
+        /// Reads a byte from the I2C device.
+        /// </summary>
+        /// <returns>A byte read from the I2C device.</returns>
+        public override byte ReadByte()
+        {
+            byte[] toRead = new byte[1];
+            Read(toRead);
+            return toRead[0];
+        }
+
+        /// <summary>
+        /// Writes data to the I2C device.
+        /// </summary>
+        /// <param name="buffer">
+        /// The buffer that contains the data to be written to the I2C device.
+        /// The data should not include the I2C device address.
+        /// </param>
+        public override void Write(ReadOnlySpan<byte> buffer)
+        {
+            ushort byteSent = 0;
+            var ftStatus = FtFunction.FT4222_I2CMaster_Write(_ftHandle, (ushort)_settings.DeviceAddress, buffer.ToArray(), (ushort)buffer.Length, ref byteSent);
```

Comment:
span should just work

Reply:
I changed as well the signature for the ref as above. So I'm passing the pointer directly